scripts/memory_logger.py

```diff
@@ -1003,15 +1034,22 @@ def verifyArgs(args):
     else:
       args.outfile = [os.getcwd() + '/' + args.run[0].replace('..', '').replace('/', '').replace(' ', '_') + '.log']
 
-  if args.pstack:
-    if platform.platform(0, 1).split('-')[:-1][0].find('Darwin') != -1:
-      try:
-        import lldb
-      except ImportError:
-        print 'Unable to import lldb. The lldb API is now supplied by \nXcode but not automatically set in your PYTHONPATH. \nPlease search the internet for how to do this if you \nwish to use --pstack on Mac OS X.\n\nNote: If you installed Xcode to the default location of \n/Applications, you should only have to perform the following:\n\n\texport PYTHONPATH=/Applications/Xcode.app/Contents/SharedFrameworks/LLDB.framework/Resources/Python:$PYTHONPATH\n'
-        sys.exit(1)
+  if args.pstack and (args.read is None and args.plot is None):
+    if args.debugger is not None:
+      if args.debugger[0] == 'lldb':
+        if platform.platform(0, 1).split('-')[:-1][0].find('Darwin') != -1:
```

Comment:
Oh and if you mean complex as in all the if statements before that... I agree. I couldn't figure out a more elegant solution unfortunately. My test case:

Some Machine  --> graphical plotting capabilities only, has no debugger available, but wishes to view debugging results that was generated on a different machine that does have a debugger. Or perhaps this machine does have a debugger available, maybe LLDB or GDB or both. But if on Darwin... sorry; gotta stick to LLDB API.